pinot-common/src/main/antlr4/org/apache/pinot/pql/parsers/PQL2.g4

```diff
@@ -78,6 +78,7 @@ predicate:
   | isClause                              # IsPredicate
   | regexpLikeClause                      # RegexpLikePredicate
   | textMatchClause                       # TextMatchPredicate
+  | jsonMatchClause                       # JsonMatchPredicate
```

Comment:
We need to support this in PQL because the query engine always take the filter from the PQL side (currently the filter optimizer only works on PQL), and we need to convert SQL filter into PQL filter and store it in `BrokerRequest`

Reply:
we can clean this up once we deprecate PQL completely